core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,88 @@
+    private urlInterpolationService: UrlInterpolationService,
+    private http: HttpClient
+  ) {}
+  /**  TODO(srijanreddy98): for some reason not able to import types (
+  * OPPORTUNITY_TYPE_SKILL,
+  * OPPORTUNITY_TYPE_TRANSLATION,
+  * OPPORTUNITY_TYPE_VOICEOVER
+  * )from constants.ts in assets */
+  _getOpportunityFromDict(opportunityType, opportunityDict) {
+    if (
+      opportunityType === 'voiceover' ||
+      opportunityType === 'translation') {
+      return new ExplorationOpportunitySummary(opportunityDict.id,
+        opportunityDict.topic_name, opportunityDict.story_title,
+        opportunityDict.chapter_title, opportunityDict.content_count,
+        opportunityDict.translation_counts);
+    } else if (opportunityType === 'skill') {
+      return new SkillOpportunity(
+        opportunityDict.id, opportunityDict.skill_description,
+        opportunityDict.topic_name, opportunityDict.question_count);
+    }
+  }
+
+  _fetchOpportunities(opportunityType, params, successCallback) {
+    this.http.get(this.urlInterpolationService.interpolateUrl(
+      this.urlTemplate, { opportunityType }
+    ), {
+      params
+    }).toPromise().then((data: any) => {
+      let opportunities = [];
+      for (let index in data.opportunities) {
+        opportunities.push(this._getOpportunityFromDict(
+          opportunityType, data.opportunities[index]));
       }
-    };
-    var _fetchOpportunities = function(
-        opportunityType, params, successCallback) {
-      return $http.get(
-        UrlInterpolationService.interpolateUrl(
-          urlTemplate, {opportunityType: opportunityType}
-        ), {
-          params: params
-        }).then(function(response) {
-        var data = response.data;
-        var opportunities = [];
-        for (var index in data.opportunities) {
-          opportunities.push(_getOpportunityFromDict(
-            opportunityType, data.opportunities[index]));
-        }
+      if (successCallback) {
         successCallback(opportunities, data.next_cursor, data.more);
-      });
-    };
-    return {
-      fetchSkillOpportunities: function(cursor, successCallback) {
-        var params = {
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_SKILL, params, successCallback);
-      },
-      fetchTranslationOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_TRANSLATION, params, successCallback);
-      },
-      fetchVoiceoverOpportunities: function(
-          languageCode, cursor, successCallback) {
-        var params = {
-          language_code: languageCode,
-          cursor: cursor
-        };
-        return _fetchOpportunities(
-          OPPORTUNITY_TYPE_VOICEOVER, params, successCallback);
       }
+    }, (err) => {
+      // console.log(err);
```

Comment:
if the reject callback is nothing being used please remove it.

Reply:
@marianazangrossi  please ignore the code in contribution-opportunities-backend-api.service.ts. I had pushed my first two commits (of a different task) into the develop branch in my fork and because of that, they appeared here. I created this branch from develop so contribution-opportunities-backend-api.service.ts showed up here.

I didn't rebase it because of this (#8763) which when goes through will replace the content of contribution-opportunities-backend-api.service. So, #8763 is expected to go in first, after which I'll update this PR with develop and then this will go in, right?

Reply:
Just a quick note, @srijanreddy98 -- please don't rebase, in general. Follow the instructions in the [developer guidelines](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change). Thanks!

Reply:
@seanlip Sure. Although I didn't find any restrictions on rebasing in the guidelines. The point which mentions not to force push kind of covers it.